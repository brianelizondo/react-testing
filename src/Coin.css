.Coin
.Coin-image{
    max-width: 100px;
}