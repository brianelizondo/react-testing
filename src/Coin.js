import React from "react";


function Coin(props) {
    const { src, face, isShow } = props;
    return (
        <div className="Coin">
            {isShow && <img className="Coin-image" src={src} alt={face} /> }
        </div>
    );
}

export default Coin;