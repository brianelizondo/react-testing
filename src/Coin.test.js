import React from "react";
import { render } from "@testing-library/react";
import Coin from "./Coin";

// Smoke Tests
it("'Coin' component renders without crashing", function() {
    render(<Coin />);
});

// Snapshot Tests
it("'Coin' matches snapshot", function() {
    const {asFragment} = render(<Coin />);
    expect(asFragment()).toMatchSnapshot();
});