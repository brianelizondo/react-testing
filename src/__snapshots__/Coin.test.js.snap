// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'Coin' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  />
</DocumentFragment>
`;
